session/session.go

```diff
@@ -2431,3 +2439,35 @@ func (s *session) recordOnTransactionExecution(err error, counter int, duration
 		}
 	}
 }
+
+func (s *session) checkPlacementPolicyBeforeCommit() error {
+	var err error
+	txnScope := s.GetSessionVars().TxnCtx.TxnScope
+	if txnScope == "" {
+		txnScope = config.DefTxnScope
+	}
+	if txnScope != config.DefTxnScope {
+		is := infoschema.GetInfoSchema(s)
+		for physicalTableID := range s.GetSessionVars().TxnCtx.TableDeltaMap {
+			bundle, ok := is.BundleByName(placement.GroupID(physicalTableID))
+			if !ok {
+				err = ddl.ErrInvalidPlacementPolicyCheck.GenWithStackByArgs(
+					fmt.Sprintf("table or partition %v don't have rule bundle with txnScope %v",
+						physicalTableID, txnScope))
+				break
+			}
+			dcLocation, ok := placement.GetLeaderDCByBundle(bundle, placement.DCLabelKey)
+			if !ok {
+				err = ddl.ErrInvalidPlacementPolicyCheck.GenWithStackByArgs(
+					fmt.Sprintf("bundle %v's leader placement policy is not defined", bundle.ID))
```

Comment:
```suggestion
					fmt.Sprintf("table or partition %v's leader placement policy is not defined", physicalTableID))
```
Users don't know what bundle is.